test/integration/scheduler_perf/config/performance-config.yaml

```diff
@@ -0,0 +1,267 @@
+- desc: Scheduling
```

Comment:
We have to duplicate the test cases by only changing the number of nodes/pods. I suggest that we define a new `basicTestCase` that uses the `TestCase` as template and overrides number of nodes/pods. This way we make the `TestCase` generic and flexible, while for most test cases we should just need to use `basicTestCase`.

```
type basicTestCase struct {
   template TestCase
   params []testParams
}

type testParams struct {
  numNodes int
  numInitPods int
  numPodsToSchedule int
}
```

Reply:
Good idea~ But how do we support both TestCase and  simpleTestCases in a YAML config? I think we need a common struct to represent all of test cases, right?

Reply:
Good question. My idea is to support simpleTestCase for now (but have it backed by the generic TestCase to not block future extension). We can always add YAML support for the generic TestCase later.

Reply:
Thx, I see

Reply:
I checked out your code and ran. Then I realized that golang benchmark will run the test multiple times, which is not what we want. I can see 2 options:
1. Forget about `testing.B` or `testing.T` and write a main function to invoke our tests. 
Pros: Not to be messed up with normal integration test runs.
Cons: Lose some nice and familiar flags with golang tests such as `test.run`.

2. Use `testing.T` with golang build tags. The idea is to use a special build tag such as `// +build scheduler_perf` that makes sure the tests are only invoked when this special tag is supplied.
Pros: Continue to use familiar golang testing utilities as well as k8s test scripts. Avoid running scheduler benchmark in normal integration tests script runs.
Cons: No

I tried option 2 a little bit but had no luck... But in theory it should work.

cc @ingvagabund

Reply:
I did some tests, the benchmark will only run one time, the default value of b.N will be 1 for our test cases.

Reply:
That's not what I see on my machine. I guess it also has something to do with the test time. Maybe if it takes too long, golang benchmark won't adjust b.N. 

Are you on Slack? Let's discuss this and some other open questions quickly on Slack.

Reply:
If possible, I prefer to use `testing.T`. Also, if using build tags in go tests works as for the ordinary go files, +10 for it. Additionally, if we want to use etcd initialization from https://github.com/ingvagabund/kubernetes/blob/scheduler-perf-rework-benchmark-framework/test/integration/scheduler_perf/main_test.go#L26, we need `testing.T`.

Reply:
> I did some tests, the benchmark will only run one time, the default value of b.N will be 1 for our test cases.

> That's not what I see on my machine. I guess it also has something to do with the test time. Maybe if it takes too long, golang benchmark won't adjust b.N.

The default benchmark time is 1s. As long as a single iteration takes more than 1s, there's going to be only one iteration.

Reply:
Yeah, @notpad just taught me this. I tried `benchtime=1x` which supposed to run the test exactly once, however for me it always ran twice... Then I tried `benchtime=1ns`, which always run the test exactly once (nobody's machine can run the test within 1ns, right?).

Reply:
For the record, some tests do finish within 1s on my machine, so we need to set a much lower threshold to accommodate everyone.

Reply:
> If possible, I prefer to use testing.T

If we get the build tags working, yes. If not, we can set the `benchtime`  to ensure tests run exactly once.